src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx

```diff
@@ -0,0 +1,283 @@
+ or see $GENIE/LICENSE
+
+ Authors: Igor Kakorin <kakorin@jinr.ru>, Joint Institute for Nuclear Research
+          Konstantin Kuzmin <kkuzmin@theor.jinr.ru >,  Joint Institute for Nuclear Research \n
+          Vadim Naumov <vnaumov@theor.jinr.ru >,  Joint Institute for Nuclear Research \n
+          based on code of Costas Andreopoulos <costas.andreopoulos \at stfc.ac.uk>
+          University of Liverpool & STFC Rutherford Appleton Lab
+
+ For the class documentation see the corresponding header file.
+
+*/
+//____________________________________________________________________________
+
+#include <sstream>
+#include <cassert>
+
+#include <TMath.h>
+#include <Math/IFunction.h>
+#include <Math/IntegratorMultiDim.h>
+
+#include "Framework/EventGen/XSecAlgorithmI.h"
+#include "Framework/ParticleData/BaryonResUtils.h"
+#include "Framework/Conventions/GBuild.h"
+#include "Framework/Conventions/Controls.h"
+#include "Framework/Conventions/Constants.h"
+#include "Framework/Conventions/Units.h"
+#include "Framework/Conventions/KinePhaseSpace.h"
+#include "Framework/Conventions/KineVar.h"
+#include "Physics/XSectionIntegration/GSLXSecFunc.h"
+#include "Framework/Interaction/Interaction.h"
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/ParticleData/PDGCodes.h"
+#include "Framework/ParticleData/PDGLibrary.h"
+#include "Physics/Resonance/XSection/MKSPPXSecWithCache.h"
+#include "Framework/Numerical/MathUtils.h"
+#include "Framework/Utils/KineUtils.h"
+#include "Framework/Utils/Cache.h"
+#include "Framework/Utils/CacheBranchFx.h"
+#include "Framework/Numerical/GSLUtils.h"
+#include "Framework/Utils/Range1.h"
+
+
+using std::ostringstream;
+
+using namespace genie;
+using namespace genie::controls;
+using namespace genie::constants;
+
+//____________________________________________________________________________
+MKSPPXSecWithCache::MKSPPXSecWithCache() :
+XSecIntegratorI()
+{
+
+}
+//____________________________________________________________________________
+MKSPPXSecWithCache::MKSPPXSecWithCache(string nm) :
+XSecIntegratorI(nm)
+{
+
+}
+//____________________________________________________________________________
+MKSPPXSecWithCache::MKSPPXSecWithCache(string nm,string conf):
+XSecIntegratorI(nm,conf)
+{
+
+}
+//____________________________________________________________________________
+MKSPPXSecWithCache::~MKSPPXSecWithCache()
+{
+
+}
+//____________________________________________________________________________
+void MKSPPXSecWithCache::CacheResExcitationXSec(const Interaction * in) const
+{
+// Cache resonance neutrino production data from free nucleons
+
+  Cache * cache = Cache::Instance();
+
+  assert(fSingleResXSecModel);
+  
+  SppChannel_t spp_channel = SppChannel::FromInteraction(in);
+
+  // Compute the number of spline knots - use at least 10 knots per decade
+  // && at least 40 knots in the full energy range
+  const double Emin       = 0.01;
```

Comment:
This needs to come from configuration

Reply:
Marco, I just follow the standard approach. The same thing is implemented in _ReinSehgalRESXSecWithCache.cxx_ and _ReinSehgalRESXSecWithCacheFast.cxx_. Then there is a need get it from configuration anywhere.